src/vcpkg/commands.build.cpp

```diff
@@ -1756,121 +1756,55 @@ namespace vcpkg
-                    Util::Vectors::append(&passthrough_env_vars, Strings::split(variable_value, ';'));
-                    break;
-                case VcpkgTripletVar::PUBLIC_ABI_OVERRIDE:
-                    public_abi_override = variable_value.empty() ? nullopt : Optional<std::string>{variable_value};
-                    break;
-                case VcpkgTripletVar::LOAD_VCVARS_ENV:
-                    if (variable_value.empty())
-                    {
-                        load_vcvars_env = !external_toolchain_file.has_value();
-                    }
-                    else
-                    {
-                        load_vcvars_env = from_cmake_bool(variable_value, kv.first).value_or_exit(VCPKG_LINE_INFO);
-                    }
-                    break;
-                case VcpkgTripletVar::DISABLE_COMPILER_TRACKING:
-                    if (variable_value.empty())
-                    {
-                        disable_compiler_tracking = false;
-                    }
-                    else
-                    {
-                        disable_compiler_tracking =
-                            from_cmake_bool(variable_value, kv.first).value_or_exit(VCPKG_LINE_INFO);
-                    }
-                    break;
-                case VcpkgTripletVar::XBOX_CONSOLE_TARGET:
-                    if (!variable_value.empty())
-                    {
-                        target_is_xbox = true;
-                    }
-                    break;
-                case VcpkgTripletVar::Z_VCPKG_GameDKLatest:
-                    if (!variable_value.empty())
-                    {
-                        gamedk_latest_path.emplace(variable_value);
-                    }
-                    break;
-            }
+            Util::Vectors::append(&passthrough_env_vars, Strings::split(*value, ';'));
+        }
+
+        Util::assign_if_set_and_nonempty(public_abi_override, cmakevars, CMakeVariablePublicAbiOverride);
+        // Note that this value must come after CMakeVariableChainloadToolchainFile because its default depends upon it
+        load_vcvars_env = !external_toolchain_file.has_value();
+        if (auto value = Util::value_if_set_and_nonempty(cmakevars, CMakeVariableLoadVcvarsEnv))
+        {
+            load_vcvars_env = from_cmake_bool(*value, CMakeVariableLoadVcvarsEnv).value_or_exit(VCPKG_LINE_INFO);
+        }
+
+        if (auto value = Util::value_if_set_and_nonempty(cmakevars, CMakeVariableDisableCompilerTracking))
+        {
+            disable_compiler_tracking =
+                from_cmake_bool(*value, CMakeVariableDisableCompilerTracking).value_or_exit(VCPKG_LINE_INFO);
         }
+
+        if (Util::value_if_set_and_nonempty(cmakevars, CMakeVariableXBoxConsoleTarget) != nullptr)
+        {
+            target_is_xbox = true;
+        }
+
+        Util::assign_if_set_and_nonempty(gamedk_latest_path, cmakevars, CMakeVariableZVcpkgGameDKLatest);
```

Comment:
Maybe separate side effects from the `value_if_set_and_nonempty` and make it a two step process?

Reply:
The reason they're combined here is to preserve the table-like structure for the fields where that is actually possible